packages/devtools_app/lib/src/timeline/flutter/flutter_frames_chart.dart

```diff
@@ -4,18 +4,74 @@
 
 import 'package:flutter/material.dart';
 
-class FlutterFramesChart extends StatelessWidget {
+import '../../flutter/controllers.dart';
+import '../../ui/fake_flutter/_real_flutter.dart';
+import '../timeline_controller.dart';
+import '../timeline_model.dart';
+
+class FlutterFramesChart extends StatefulWidget {
+  @override
+  _FlutterFramesChartState createState() => _FlutterFramesChartState();
+}
+
+class _FlutterFramesChartState extends State<FlutterFramesChart> {
+  TimelineController _controller;
+
+  List<TimelineFrame> frames = [];
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _controller = Controllers.of(context).timeline;
```

Comment:
Suggestion: rather than using StreamBuilder below, just listen to the stream and call setState() with the updated data.

That will save you the trouble of managing the async snapshot.

```dart
didChangeDependencies() {
  _controller.frameBasedTimeline.onFrameAdded.listen((extraFrames) {
    setState(() {
      frames.add(extraFrames);
    });
  });
}
```

Right now, this state change is happening silently in build, which is a code smell.